### DISCUSSION TITLE
is there any zfs tools command like xfs_db to trace zfs meta

### DISCUSSION THREAD

**Discussion by httaotao** [openzfs/zfs#12982] [Q&A]
hello, I am not studying  zfs.  These days, I am curious  zfs  whether has the command to trace the zpool meta , like the xfs command ``` xfs_db ```

the command  likes that
```
xfs_db -c "agfl 0" -c "p" /dev/xxx
```

this command can trace the ```agfl``` in xfs.  ```agfl``` is AG internel free list (AG in xfs  like metaslab in zfs), which is a special b+ tree.

thanks.

**Comment by rincebrain** [comment]
Some kind of [zpool debugging](https://openzfs.github.io/openzfs-docs/man/8/zdb.8.html) tool, you say.
